src/panels/config/integrations/config-entry/ha-config-entry-page.ts

```diff
@@ -145,26 +146,22 @@ class HaConfigEntryPage extends LitElement {
   }
 
   private _removeEntry() {
-    if (
-      !confirm(
-        this.hass.localize(
-          "ui.panel.config.integrations.config_entry.delete_confirm"
-        )
-      )
-    ) {
-      return;
-    }
-
-    deleteConfigEntry(this.hass, this.configEntryId).then((result) => {
-      fireEvent(this, "hass-reload-entries");
-      if (result.require_restart) {
-        alert(
-          this.hass.localize(
-            "ui.panel.config.integrations.config_entry.restart_confirm"
-          )
-        );
-      }
-      navigate(this, "/config/integrations/dashboard", true);
+    showConfirmationDialog(this, {
+      text: this.hass.localize(
+        "ui.panel.config.integrations.config_entry.delete_confirm"
+      ),
+      confirm: () =>
+        deleteConfigEntry(this.hass, this.configEntryId).then((result) => {
+          fireEvent(this, "hass-reload-entries");
+          if (result.require_restart) {
+            alert(
+              this.hass.localize(
+                "ui.panel.config.integrations.config_entry.restart_confirm"
+              )
+            );
+          }
+          navigate(this, "/config/integrations/dashboard", true);
+        }),
```

Comment:
Move this to a separate function like you did for the rest:
`confirm: () => this._deleteConfigEntry()`